### PR TITLE
Add plugin for SBOM generation

### PR DESCRIPTION
**Pull request by ivargrimstad**
Also update plugin versions in pluginManagement secion

### CODE REVIEW

File: parent/pom.xml [hunk 1]
```diff
@@ -167,16 +167,46 @@
...
+                <plugin>
+                    <groupId>org.asciidoctor</groupId>
+                    <artifactId>asciidoctor-maven-plugin</artifactId>
+                    <version>2.2.4</version>
                 </plugin>
             </plugins>
         </pluginManagement>
+
+        <plugins>
+            <plugin>
```
Comment (pzygielo) [inline parent/pom.xml:192]: Not sure about the change. The plugin will run in every module using this as parent, and for example in OpenMQ I see this:
```
[INFO] --- cyclonedx-maven-plugin:2.7.9:makeAggregateBom (default) @ mq ---
[INFO] CycloneDX: Resolving Dependencies
[INFO] CycloneDX: Creating BOM version 1.4 with 1 component(s)
[INFO] CycloneDX: Writing and validating BOM (XML): .../mq/main/target/bom.xml
[INFO]            attaching as mq-6.5.0-SNAPSHOT-cyclonedx.xml
[INFO] CycloneDX: Writing and validating BOM (JSON): .../mq/main/target/bom.json
[WARNING] Unknown keyword additionalItems - you should define your own Meta Schema. If the keyword is irrelevant for validation, just use a NonValidationKeyword
[INFO]            attaching as mq-6.5.0-SNAPSHOT-cyclonedx.json
```

Comment (pzygielo) [reply parent/pom.xml:192]: Is this some EF requirement? I don't understand why every child project is going to need this.

Comment (ivargrimstad) [reply parent/pom.xml:192]: Not at the moment, but there will eventually be a requirement to produce SBOMs coming up in the near future. This work is to be upfront with that. That said, if it generates a lot of problems for the projects, we can add it in a profile that can be disabled for those projects experiencing problems until it is fixed.

**Review by edburns — CHANGES_REQUESTED** [review]
Wouldn't you also need to increment the version of the POM itself, in line 27?

**Comment by pzygielo** [comment]
> Wouldn't you also need to increment the version of the POM itself, in line 27?

No need to do so, because target branch is already updated in that place.

**Comment by starksm64** [comment]
> Wouldn't you also need to increment the version of the POM itself, in line 27?

That is automatically managed by the release plugin. It is only updated when there is an actual release, not on every commit.

**Review by starksm64 — APPROVED** [review]

**Comment by starksm64** [comment]
I tried this on the cdi project and saw no ill effects

**Comment by ivargrimstad** [comment]
> I tried this on the cdi project and saw no ill effects

Thanks, Scott!

**Pull request merged**